services/symfony/symfony-insight.service.js

```diff
@@ -0,0 +1,277 @@
+        // the xml nodes to always be parsed as an array. Currently, if the response
+        // only contains a single violation then it will be parsed as an object,
+        // otherwise it will be parsed as an array.
+        violation: [
+          violationSchema,
+          Joi.array()
+            .items(violationSchema)
+            .required(),
+        ],
+      }),
+    }),
+  }).required(),
+}).required()
+
+const keywords = ['sensiolabs']
+
+module.exports = class SymfonyInsight extends BaseXmlService {
+  static render({
+    metric,
+    status,
+    grade,
+    numViolations,
+    numCriticalViolations,
+    numMajorViolations,
+    numMinorViolations,
+    numInfoViolations,
+  }) {
+    if (status !== 'finished') {
+      return {
+        message: 'pending',
+        color: 'grey',
+      }
+    }
+
+    if (metric === 'grade') {
+      return this.renderGradeBadge({ grade })
+    } else {
+      return this.renderViolationsBadge({
+        numViolations,
+        numCriticalViolations,
+        numMajorViolations,
+        numMinorViolations,
+        numInfoViolations,
+      })
+    }
+  }
+
+  static renderGradeBadge({ grade }) {
+    let color,
+      message = grade
+    if (grade === 'platinum') {
+      color = '#E5E4E2'
+    } else if (grade === 'gold') {
+      color = '#EBC760'
+    } else if (grade === 'silver') {
+      color = '#C0C0C0'
+    } else if (grade === 'bronze') {
+      color = '#C88F6A'
+    } else {
+      message = 'no medal'
+      color = 'red'
+    }
+
+    return {
+      message,
+      color,
+    }
+  }
+
+  static renderViolationsBadge({
+    numViolations,
+    numCriticalViolations,
+    numMajorViolations,
+    numMinorViolations,
+    numInfoViolations,
+  }) {
+    if (numViolations === 0) {
+      return {
+        message: '0',
+        color: 'brightgreen',
+      }
+    }
+
+    let color = 'yellowgreen'
+    const violationSummary = []
+
+    if (numInfoViolations > 0) {
+      violationSummary.push(`${numInfoViolations} info`)
+    }
+    if (numMinorViolations > 0) {
+      violationSummary.unshift(`${numMinorViolations} minor`)
+      color = 'yellow'
+    }
+    if (numMajorViolations > 0) {
+      violationSummary.unshift(`${numMajorViolations} major`)
+      color = 'orange'
+    }
+    if (numCriticalViolations > 0) {
+      violationSummary.unshift(`${numCriticalViolations} critical`)
+      color = 'red'
+    }
+
+    return {
+      message: violationSummary.join(', '),
+      color,
+    }
+  }
+
+  static get category() {
+    return 'quality'
+  }
+
+  static get route() {
+    return {
+      base: '',
+      // The SymfonyInsight service was previously branded as SensioLabs, and
+      // accordingly the badge path used to be /sensiolabs/i/projectUuid'.
+      // This is used to provide backward compatibility for the old path as well as
+      // supporting the new/current path.
+      format: '(?:sensiolabs/i|symfony/i/(grade|violations))/([^/]+)',
```

Comment:
`violations` is the new path, right? After looking at the example badge, I'm wondering if it should be labeled `checks` instead.

Reply:
I actually think it should be `violations` or `alerts`. From what I've gathered from the Symfony docs they refer to `checks` as the things their analysis performs (the sorts of conditions things they scan/check for, etc.), so their "checks" are fairly synonymous to "rules" in my mind.

Reply:
Okay. So the URL will stay `violations`. Should we change the label from **checks**  to **violations**, then?

Reply:
that's what I've done locally 😄, getting ready to push